### ISSUE TITLE
Make it more clear that SourceRef is for cluster

### ISSUE THREAD

**Issue by johanandren** [akka/akka#26173]
Using SourceRef between remoting nodes without cluster is a bad fit since it uses remote deathwatch and if the failure detector hits it will cause quarantining. 

The docs says "allow running Akka Streams across multiple nodes within an Akka Cluster." https://doc.akka.io/docs/akka/current/stream/stream-refs.html#introduction but we can perhaps make it more explicit that using it with remoting directly is dangerous?

**Comment by patriknw** [comment]
I agree, it is important to only use them within a Cluster, or be prepared of the consequences.
